org.adempiere.base/src/org/compiere/report/MReportSource.java

```diff
@@ -241,7 +241,29 @@ private String getWhereCombination(int PA_Hierarchy_ID) {
 		} else
 			if (isIncludeNullsUserElement2())
 				whcomb.append(" AND UserElement2_ID IS NULL");
-		
+
+		if (getUser1_ID() > 0)
+		{
+			String whtree = "User1_ID=" + getUser1_ID(); // No Tree
+			if (isIncludeNullsUserList1())
+				whcomb.append(" AND (User1_ID IS NULL OR ").append(whtree).append(")");
+			else
+				whcomb.append(" AND ").append(whtree);
+		}
+		else if (isIncludeNullsUserList1())
+			whcomb.append(" AND User1_ID IS NULL");
+
+		if (getUser2_ID() > 0)
+		{
+			String whtree = "User2_ID=" + getUser2_ID(); // No Tree
+			if (isIncludeNullsUserList2())
+				whcomb.append(" AND (User2_ID IS NULL OR ").append(whtree).append(")");
+			else
+				whcomb.append(" AND ").append(whtree);
+		}
+		else if (isIncludeNullsUserList2())
+			whcomb.append(" AND User2_ID IS NULL");
+

```

Comment:
With this change, at DB level, both report line id and report column id is nullable. Hence, I think it is necessary to verify here that either report line id or report column id is > 0.